src/main/java/com/uniqueAuction/domain/trade/repository/SaleRepositoryImpl.java

```diff
@@ -1,18 +1,30 @@
 package com.uniqueAuction.domain.trade.repository;
 
-import com.uniqueAuction.domain.trade.entity.Sale;
-import org.springframework.stereotype.Repository;
-
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.atomic.AtomicLong;
 
+import org.springframework.stereotype.Repository;
+
+import com.uniqueAuction.domain.trade.entity.Sale;
+
 @Repository
 public class SaleRepositoryImpl implements SaleRepository {
-    private static final ConcurrentHashMap<Long, Sale> sales = new ConcurrentHashMap<>();
-    private static final AtomicLong sequence = new AtomicLong();
+	private static final ConcurrentHashMap<Long, Sale> sales = new ConcurrentHashMap<>();
+	private static final AtomicLong sequence = new AtomicLong();
+
+	@Override
+	public Long save(Sale sale) {
+		sales.put(sequence.addAndGet(1), sale);
+		return sequence.addAndGet(1);
+	}
 
-    @Override
-    public void save(Sale sale) {
-        sales.put(sequence.addAndGet(1) , sale);
-    }
+	@Override
+	public Long isExists(String modelNumber, String size) {
+		return sales.entrySet()
+			.stream()
+			.filter(e -> e.getValue().getModelNumber().equals(modelNumber))
+			.filter(e -> e.getValue().getSize().equals(size)).findFirst()
+			.map(e -> e.getValue().getId())
+			.orElse(0L);
```

Comment:
아래 관점으로 개선한 코드입니다. 

- exists 함수명 수정
- exists는 존재 여부만 반환하기때문에 Boolean 으로 타입 변경
- filter는 두번 할 필요없음, 그리고 하나만 찾으면 더 찾을 필요도 없음

```suggestion
	public boolean exists(String modelNumber, String size) {
        return sales.values().stream()
                .anyMatch(e -> e.getModelNumber().equals(modelNumber) && e.getSize().equals(size))
    }
```

Reply:
trade 객체에 id를 저장할 필요가 있어 존재 여부 함수에서 getSaleId, getPurchaseId와 같은 형삭으로 변경하였습니다.